desktop/app/first-time.js

```diff
@@ -0,0 +1,36 @@
+// Checks to see if we've launched before (we touch started.txt). If not, write the file and open a browser
+import {splashRoot} from '../shared/constants/platform.native.desktop'
+import path from 'path'
+import fs from 'fs'
+
+const filePath = path.join(splashRoot, 'started.txt')
+
+let _firstTimeCachedValue = null
+
+let isFirstTime: () => Promise<boolean> = () => new Promise((resolve, reject) => {
```

Comment:
yeah so its worth reading stuff like http://www.mattgreer.org/articles/promises-in-wicked-detail/